vicky/src/lib/vicky/scheduler.rs

```diff
@@ -343,8 +355,65 @@ mod tests {
                 .build(),
         ];
 
-        let res = Scheduler::new(tasks, &[]).unwrap();
+        let res = Scheduler::new(tasks, &[], &[]).unwrap();
         // Test 1 is currently running and has the write lock
         assert_eq!(res.get_next_task(), None)
     }
+
+    #[test]
+    fn schedule_with_poisoned_lock() {
+        let tasks = vec![Task::builder()
+            .with_display_name("I need to do something")
+            .with_write_lock("Entire Prod Cluster")
+            .build()];
+        let poisoned_locks = vec![Lock::WRITE {
+            name: "Entire Prod Cluster".to_string(),
+            poisoned: Some(Uuid::new_v4()),
+        }];
+
+        let res = Scheduler::new(tasks, &poisoned_locks, &[]).unwrap();
+
+        assert_eq!(res.get_next_task(), None);
+    }
+
+    #[test]
+    fn schedule_different_tasks_with_poisoned_lock() {
+        let tasks = vec![
+            Task::builder()
+                .with_display_name("I need to do something")
+                .with_write_lock("Entire Prod Cluster")
+                .build(),
+            Task::builder()
+                .with_display_name("I need to test something")
+                .with_write_lock("Entire Staging Cluster")
+                .build(),
+        ];
+        let poisoned_locks = vec![Lock::WRITE {
+            name: "Entire Prod Cluster".to_string(),
+            poisoned: Some(Uuid::new_v4()),
+        }];
+
+        let res = Scheduler::new(tasks, &poisoned_locks, &[]).unwrap();
+
+        assert_eq!(
+            res.get_next_task().unwrap().display_name,
+            "I need to test something"
+        );
+    }
+
+    #[test]
+    fn schedule_different_tasks_with_poisoned_lock_ro() {
```

Comment:
This is the same test as `schedule_with_poisoned_lock`

Reply:
nope

Reply:
a read lock can be scheduled with N other read locks, but not when one other poisoned read lock exists, while write locks shouldn't be scheduled together either way.